src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,169 @@
+package duke;
+
+import duke.Exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.IOException;
+import java.time.DateTimeException;
+
+/**
+ *
+ * Interpretes and executes the inputs that the user passes in.
+ */
+public class Parser {
+    public static Ui ui = new Ui();
+
+
+    /**
+     * Deals with user inputs beginning with mark, unmark,
+     * delete, list.
+     * It also deals with what happens when the user types in an unrecognised
+     * command.
+     *
+     * @param input what the user is typing in.
+     * @param storage the storage that is being used.
+     * @param tasks the TaskList that is being used to store the tasks.
+     * @throws DukeException
+     * @throws NumberFormatException
+     */
+
+    public static void userCommand(String input, Storage storage, TaskList tasks) throws DukeException, NumberFormatException {
+        try {
+            if (input.startsWith("mark")) {
+                int taskIndex = Integer.parseInt(input.substring(5));
+                if (tasks.getTask(taskIndex - 1).getStatusIcon() == "X") {
+                    throw new DukeException("\tThis task has already been marked as done!");
+                }
+                tasks.markTaskAsDone(taskIndex - 1);
+                storage.writeTasksToFile(tasks);
+                ui.printMarkTasksAsDone(taskIndex, tasks);
```

Comment:
Perhaps there could be more spacing to make this readable? Not sure I like the lines clumped up together but that could be a personal preference too. Similar with the following lines.

```suggestion
                int taskIndex = Integer.parseInt(input.substring(5));
                
                if (tasks.getTask(taskIndex - 1).getStatusIcon() == "X") {
                    throw new DukeException("\tThis task has already been marked as done!");
                }
                
                tasks.markTaskAsDone(taskIndex - 1);
                storage.writeTasksToFile(tasks);
                ui.printMarkTasksAsDone(taskIndex, tasks);
```